src/com/google/enterprise/adaptor/DocumentContentTransformerPipeline.java

```diff
@@ -0,0 +1,70 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The document content transformer pipeline holds all document content transformers
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransformerPipeline {
+
+  private static final Logger LOG =
+      Logger.getLogger(DocumentContentTransformerPipeline.class.getName());
+
+  private List<Map<String, String>> contentTransformers;
+
+  public DocumentContentTransformerPipeline(final List<Map<String, String>> contentTransformers) {
+    this.contentTransformers = contentTransformers;
+  }
+
+  /**
+   * Creates a new transformer pipeline.
+   *
+   * @param original    original content stream
+   * @param contentType content type
+   * @param metadata    metadata
+   * @return stream pipeline or original output stream
+   */
+  public final OutputStream createPipeline(final OutputStream original,
```

Comment:
I see what you mean about there not needing to be call to transform().  It is because the OuputStream is wrapped and the pipeline invocation is implicit in the layering of OutputStream rather than needing an explicit call.  

Then perhaps this class is not necessary or perhaps it is a bit misnamed at least.  That is, consider that if the instance of this class is a "DocumentContentTransformerPipeline" then it should not have a method to "createPipeline" as it is already is a pipeline by virtue of its name.

Let me think of some better naming by tmrw hopefully.